liquibase-core/src/main/java/liquibase/diff/output/changelog/DiffToChangeLog.java

```diff
@@ -91,23 +92,38 @@ public void print(PrintStream out) throws ParserConfigurationException, IOExcept
 
     public void print(String changeLogFile, ChangeLogSerializer changeLogSerializer) throws ParserConfigurationException, IOException, DatabaseException {
         this.changeSetPath = changeLogFile;
-        File file = new File(changeLogFile);
+        final PathHandlerFactory pathHandlerFactory = Scope.getCurrentScope().getSingleton(PathHandlerFactory.class);
+        Resource resource = pathHandlerFactory.getResource(changeLogFile, true);
+        boolean fileExists;
+        // todo, an exists method would be useful here, instead of this hack
+        try (InputStream is = resource.openInputStream()) {
+            fileExists = is != null;
+        } catch (IOException | NullPointerException e) {
+            fileExists = false;
+            Scope.getCurrentScope().getLog(getClass()).warning("Unable to open input stream to file " + changeLogFile, e);
+        }
 
         final Map<String, Object> newScopeObjects = new HashMap<>();
 
-        File objectsDir = null;
+        String objectsDir = null;
+        String parent;
+        if (resource != null) {
+            parent = CommonsFilenameUtils.getPath(resource.getPath());
+        } else {
+            parent = CommonsFilenameUtils.getPath(changeLogFile);
+        }
         if (changeLogFile.toLowerCase().endsWith("sql")) {
             DeprecatedConfigurationValueProvider.setData("liquibase.pro.sql.inline", "true");
         } else if (this.diffResult.getComparisonSnapshot() instanceof EmptyDatabaseSnapshot) {
-            objectsDir = new File(file.getParentFile(), "objects");
+            objectsDir = CommonsFilenameUtils.concat(parent, "objects");
         } else {
-            objectsDir = new File(file.getParentFile(), "objects-" + new Date().getTime());
+            objectsDir = CommonsFilenameUtils.concat(parent, "objects-" + new Date().getTime());
         }
 
         if (objectsDir != null) {
-            if (objectsDir.exists()) {
-                throw new UnexpectedLiquibaseException("The generatechangelog command would overwrite your existing stored logic files. To run this command please remove or rename the '"+objectsDir.getCanonicalPath()+"' dir in your local project directory");
-            }
+//            if (objectsDir.exists()) {
+//                throw new UnexpectedLiquibaseException("The generatechangelog command would overwrite your existing stored logic files. To run this command please remove or rename the '"+objectsDir.getCanonicalPath()+"' dir in your local project directory");
+//            }
```

Comment:
I can't think of a good way to do this check.

Reply:
Wes thinks that this is possible with the aws cli

s3 ls bucket:foldername (without slash) - won't be found
s3 ls bucket:foldername (with slash) - will respond yes, that a directory does exist here